services/workers/src/test/java/org/sagebionetworks/download/worker/DownloadListWorkerIntegrationTest.java

```diff
@@ -221,6 +249,350 @@ public void testAddToDownloadListWorkerWithFolder() throws Exception {
+		}, MAX_WAIT_MS);		
+		
+		// We create a snapshot
+		TableUpdateTransactionRequest transactionRequest = new TableUpdateTransactionRequest();
+		transactionRequest.setEntityId(dataset.getId());
+		transactionRequest.setCreateSnapshot(true);
+		transactionRequest.setSnapshotOptions(new SnapshotRequest().setSnapshotComment("snapshot"));
+		
+		TableUpdateTransactionResponse snapshotResponse = asynchronousJobWorkerHelper.assertJobResponse(adminUser, transactionRequest, (TableUpdateTransactionResponse response) -> {
+			assertNotNull(response.getSnapshotVersionNumber());
+		}, MAX_WAIT_MS).getResponse();
+
+		asynchronousJobWorkerHelper.assertQueryResult(adminUser, "SELECT * FROM " + dataset.getId()  + "." + snapshotResponse.getSnapshotVersionNumber(), (QueryResultBundle result) -> {
+			assertEquals(2L, result.getQueryResult().getQueryResults().getRows().size());
+		}, MAX_WAIT_MS);
+
+		// Now trash one of the files
+		trashManager.moveToTrash(user, trashedFile.getId(), false);
+
+		AddToDownloadListRequest addRequest = new AddToDownloadListRequest().setQuery(new Query().setSql("SELECT * FROM " + dataset.getId() + "." + snapshotResponse.getSnapshotVersionNumber()));
+		
+		// call under test
+		asynchronousJobWorkerHelper.assertJobResponse(adminUser, addRequest, (AddToDownloadListResponse response) -> {
+			// PLFM-7263: This fails returning 2 instead of 1
+			assertEquals(1L, response.getNumberOfFilesAdded());
+		}, MAX_WAIT_MS, MAX_RETRIES);
+	}
+	
+	// Case for statistics on files when a file on the download list is deleted 
+	@Test
+	@Disabled
+	public void testStatisticsWithDeletedItem() throws Exception {
+		Node file = createFileHierarchy(ACCESS_TYPE.READ, ACCESS_TYPE.DOWNLOAD);
+		
+		// Add another file to the hierarchy, we will delete this
+		Node deletedFile = nodeDaoHelper.create((n) -> {
+			n.setParentId(file.getParentId());
+			n.setNodeType(EntityType.file);
+			n.setName("DeletedFile");
+			n.setFileHandleId(file.getFileHandleId());
+		});
+				
+		// Call under test
+		AddBatchOfFilesToDownloadListResponse result = downloadListmanager.addBatchOfFilesToDownloadList(user, new AddBatchOfFilesToDownloadListRequest().setBatchToAdd(List.of(
+			new DownloadListItem().setFileEntityId(file.getId()),
+			new DownloadListItem().setFileEntityId(deletedFile.getId())
+		)));
+
+		// Both files should have been added
+		assertEquals(2, result.getNumberOfFilesAdded());
+		
+		// We now delete the file after adding it
+		nodeDao.delete(deletedFile.getId());		
+		
+		asynchronousJobWorkerHelper.assertJobResponse(user, new DownloadListQueryRequest().setRequestDetails(new FilesStatisticsRequest()), (DownloadListQueryResponse response) -> {
+			FilesStatisticsResponse details = (FilesStatisticsResponse) response.getResponseDetails();
+			assertEquals(1, details.getNumberOfFilesAvailableForDownload());
+			// PLFM-7263: This fails returning 1 instead of 0
+			assertEquals(0, details.getNumberOfFilesRequiringAction());
```

Comment:
maybe this should be a one, since one action the user can take is to manually delete the file from their list.